hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -263,8 +263,14 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIn
       omKeyInfo.getMetadata().putAll(KeyValueUtil.getFromProtobuf(
           commitKeyArgs.getMetadataList()));
 
+      // non-null indicates it is necessary to update the open key
+      OmKeyInfo newOpenKeyInfo = null;
+
       if (isHSync) {
-        omKeyInfo.getMetadata().put(OzoneConsts.HSYNC_CLIENT_ID, clientIdString);
+        if (!OmKeyHSyncUtil.isHSyncedPreviously(omKeyInfo, clientIdString, dbOpenKey)) {
```

Comment:
newOpenKeyInfo is created after omKeyInfo updated with commit metadata args above. This might have additional change of openKeyInfo. We can avoid this updating other info.

Reply:
Moved `omKeyInfo.getMetadata().putAll` call after this `if`. This implies client-provided key info arg metadata could override the default.